src/main/java/com/comcast/zucchini/ZucchiniShutdownHook.java

```diff
@@ -117,10 +120,24 @@ public void run() {
                     rptName = System.getenv(NAME_ENV_VAR);
 
 
-                ReportBuilder reportBuilder = new ReportBuilder(pathList, html, "", rptName + version, "Zucchini" + version, true, true, true, false, false, "", false);
+                ReportBuilder reportBuilder = new ReportBuilder(pathList, html, "", rptName + version, "Zucchini" + version, 
+                        true, true, true, false, false, "", false);
                 reportBuilder.generateReports();
-
                 boolean buildResult = reportBuilder.getBuildStatus();
+                
+                if ((options != null) && (options.failureReport())) {
+                    JsonArray failedFeatures = parser.parse(features.toString()).getAsJsonArray(); // deep copy
+                    filterFailedScenarios(failedFeatures);
+                    File fjson = new File(options.failureJson());
+                    FileUtils.write(fjson, failedFeatures.toString());
+                    
+                    File failureDir = new File(options.failureHtml());
+                    failureDir.mkdirs();
+                    reportBuilder = new ReportBuilder(Arrays.asList(fjson.getAbsolutePath()), failureDir, "", rptName + version, "Zucchini" + version, 
```

Comment:
This is almost the same as lines 123-125. Please factor them out. When we'll be upgrading this dependency to the latest version (2.7.0), it'll be much simpler to maintain.